File: packages/style-analyzer/fixtures/complexity-sample.py
def add(a, b):
    return a + b


def process_data(input_list):
    results = []
    seen = set()
    errors = []
    count = 0

    for item in input_list:
        if item in seen:
            continue
        seen.add(item)

        trimmed = item.strip()
        if len(trimmed) == 0:
            errors.append("empty")
            continue

        upper = trimmed.upper()
        results.append(upper)
        count += 1

    if len(errors) > 0:
        print(errors)

    return results


def deeply_nested(data):
    if data:
        if isinstance(data, dict):
            if "items" in data:
                for item in data["items"]:
                    if isinstance(item, str):
                        return item
    return ""

File: packages/style-analyzer/fixtures/complexity-sample.ts
function add(a: number, b: number): number {
  return a + b;
}

function processData(input: string[]): string[] {
  const results: string[] = [];
  const seen = new Set<string>();
  const errors: string[] = [];
  let count = 0;

  for (const item of input) {
    if (seen.has(item)) {
      continue;
    }
    seen.add(item);

    const trimmed = item.trim();
    if (trimmed.length === 0) {
      errors.push("empty");
      continue;
    }

    const upper = trimmed.toUpperCase();
    results.push(upper);
    count++;
  }

  if (errors.length > 0) {
    console.warn(errors);
  }

  return results;
}

function deeplyNested(data: unknown): string {
  if (data) {
    if (typeof data === "object") {
      if (Array.isArray(data)) {
        for (const item of data) {
          if (typeof item === "string") {
            return item;
          }
        }
      }
    }
  }
  return "";
}

function classify(value: number): string {
  if (value < 0) {
    return "negative";
  } else if (value === 0) {
    return "zero";
  } else if (value < 10) {
    return "small";
  } else if (value < 100) {
    return "medium";
  } else {
    return "large";
  }
}
